packages/smooth_app/lib/data_models/onboarding_loader.dart

```diff
@@ -18,13 +18,20 @@ class OnboardingLoader {
   ) async {
     switch (page) {
       case OnboardingPage.WELCOME:
-        await LoadingDialog.run<void>(
+        final bool? error = await LoadingDialog.run<bool>(
           context: context,
           future: _downloadData(),
           title: AppLocalizations.of(context)
               .onboarding_welcome_loading_dialog_title,
           dismissible: false,
         );
+        if (error ?? false) {
+          await LoadingDialog.error(
+              context: context,
+              // ignore: use_build_context_synchronously
```

Comment:
Please get rid of that `ignore` and pre-compute `appLocalizations` a bit earlier instead.